### DISCUSSION TITLE
q-popup-edit: change popup-content-class on @update:model-value

### DISCUSSION THREAD

**Discussion by sideburn** [quasarframework/quasar#15219] [General - Components / Directives / etc] [answered]
Hello,

I have a Q-table with "in place" editing of certain cells in the table. I want the color of the text to change to indicate that this entry has changed. I haven't figured out a way to do this.

I figured I could somehow change popup-content-class="my-new-style-on-cange"
But I am not sure how to get this to work.

Here is a code example of what I have:

` <q-popup-edit buttons v-model="props.row.company" v-slot="scope">

    <q-select
        popup-content-class="bg-grey-9 "   // I want to add the "text-warning" style on a change event
        v-model="scope.value"
        :options="companySelectionOptions"
        @keyup.enter="scope.set" //when SET button even change text color by adding a style to popup-content-class
    > </q-select>
</q-popup-edit>`

Thank you for any help.

-Tavis

**Comment by metalsadman** [comment, marked as answer]
you can add it as property of your row object, then modify it in your event. ie.
sample row object `{ warningClass: '', ...}`
```html
<q-select
    :popup-content-class="props.row.warningClass "
    :options="companySelectionOptions"
    @keyup.enter="scope.set, scope.row.warningClass = 'text-warning'"popup-content-class
> </q-select>
```

**Reply by metalsadman** [reply]
or transfer that inline expression into a event handler function. ie. `@keyup.enter="handleChange(scope)"`

**Reply by sideburn** [reply]
Awesome, thank you. Thats what I was thinking but I just couldn't quite wrap my head around how to set the new property.

**Reply by sideburn** [reply]
had to make a few edits... for the record this is what worked:
`rows: [
      {
          warningClass: "text-red",
      },
  ],
  
  <class="props.row.warningClass">{{ col.value }}
          <q-select
                  v-model="scope.value"
                  :options="companySelectionOptions"
                  @keyup.enter="scope.set"
                  @update:model-value="handlePopupChange(props.row)"
          ></q-select>
  
  methods: {
      handlePopupChange(row) {
        row.warningClass = "text-warning";
      },
  }`

**Reply by metalsadman** [reply]
for readability apply formatting https://gist.github.com/Robert-96/048fddefc20e6860aeca6949a2443adf#fenced-code-blocks
